pennylane/decomposition/symbolic_decomposition.py

```diff
@@ -16,238 +16,166 @@
+    # base class is adjoint, and the base of the base is the target class
+    target_class, target_params = base_params["base_class"], base_params["base_params"]
+    return {
+        adjoint_resource_rep(
+            qml.ops.Pow, {"base_class": target_class, "base_params": target_params, "z": z}
+        ): 1
+    }
 
-def _adjoint_pow_resource(base_class, base_params):  # pylint: disable=unused-argument
-    """Resources of the adjoint of the power of a gate whose adjoint is of the same type."""
-    base, base_params, z = base_params["base_class"], base_params["base_params"], base_params["z"]
-    # The adjoint of the base is assumed to be of the same type as the base.
-    return {pow_resource_rep(base, base_params, z): 1}
 
+@register_resources(_flip_pow_adjoint_resource)
+def flip_pow_adjoint(*params, wires, base, z, **__):
+    """Decompose the power of an adjoint by power to the base of the adjoint and
+    then taking the adjoint of the power."""
+    _, struct = base.base._flatten()
+    new_struct = (wires, *struct[1:])
+    base_op = base.base._unflatten(params, new_struct)
+    qml.adjoint(qml.pow(base_op, z))
 
-@register_resources(_adjoint_pow_resource)
-def adjoint_pow_decomp(*_, base, **__):
-    """Decompose the adjoint of the power of a gate that has its own adjoint."""
-    qml.pow(base.base.adjoint(), z=base.z)
 
+def _pow_self_adjoint_resource(base_class, base_params, z):  # pylint: disable=unused-argument
+    if (not is_integer(z)) or z < 0:
+        raise DecompositionNotApplicable
+    return {resource_rep(base_class, **base_params): z % 2}
 
-def _pow_resource(base_class, base_params, z):
-    """Resources of the power of a gate."""
-    if not isinstance(z, int) or z < 0:
-        raise NotImplementedError("Non-integer or negative powers are not supported yet.")
-    return {resource_rep(base_class, **base_params): z}
 
+@register_resources(_pow_self_adjoint_resource)
+def pow_of_self_adjoint(*params, wires, base, z, **__):
+    """Decompose the power of a self-adjoint operator, assumes z is an integer."""
 
-@register_resources(_pow_resource)
-def repeat_pow_base(*_, base, z, **__):
-    """Decompose the power of a gate."""
-    assert isinstance(z, int) and z >= 0
-    for _ in range(z):
-        base._unflatten(*base._flatten())  # pylint: disable=protected-access
-
-
-def _pow_pow_resource(base_class, base_params, z):  # pylint: disable=unused-argument
-    """Resources of the power of the power of a gate."""
-    base_class, base_params, base_z = (
-        base_params["base_class"],
-        base_params["base_params"],
-        base_params["z"],
-    )
-    return {pow_resource_rep(base_class, base_params, z * base_z): 1}
-
-
-@register_resources(_pow_pow_resource)
-def merge_powers(*_, base, z, **__):
-    """Decompose the power of the power of a gate."""
-    qml.pow(base.base, z=z * base.z)
-
-
-@functools.lru_cache(maxsize=1)
-def same_type_adjoint_ops():
```

Comment:
This is completely removed, generalized to operator-specific adjoint decomposition rules.